File: VB/WebSite/App_Code/CustomResources.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data.OleDb
Imports System.Data

#Region "CarsDbAdapterHelper"
Public Class CarsDbAdapterHelper
	Inherits DbAdapterHelperBase
	#Region "CreateOleDbDataAdapter"
	Public Shared Function CreateOleDbDataAdapter() As OleDbDataAdapter
		Dim helper As New CarsDbAdapterHelper()
		Return helper.Adapter
	End Function
	#End Region

	#Region "CreateSelectionCommand"
	Protected Overrides Function CreateSelectionCommand(ByVal connection As OleDbConnection) As OleDbCommand
		Dim command As New OleDbCommand()
		command.Connection = connection
		command.CommandText = "SELECT ID, Model FROM Cars"
		command.CommandType = CommandType.Text
		Return command
	End Function
	#End Region
End Class
#End Region

#Region "CustomResourceDataSource"
Public Class CustomResourceDataSource
	Private adapter_Renamed As OleDbDataAdapter

	Public Sub New()
		Me.adapter_Renamed = CarsDbAdapterHelper.CreateOleDbDataAdapter()
	End Sub

	Public ReadOnly Property Adapter() As OleDbDataAdapter
		Get
			Return adapter_Renamed
		End Get
	End Property

	#Region "ObjectDataSource methods"
	Public Function SelectMethodHandler() As DataTable
		Dim table As New DataTable()
		Adapter.Fill(table)
		Return table
	End Function
	#End Region
End Class
#End Region

File: CS/WebSite/App_Code/CustomResources.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data.OleDb;
using System.Data;

#region CarsDbAdapterHelper
public class CarsDbAdapterHelper : DbAdapterHelperBase {
	#region CreateOleDbDataAdapter
	public static OleDbDataAdapter CreateOleDbDataAdapter() {
		CarsDbAdapterHelper helper = new CarsDbAdapterHelper();
		return helper.Adapter;
	}
	#endregion

	#region CreateSelectionCommand
	protected override OleDbCommand CreateSelectionCommand(OleDbConnection connection) {
		OleDbCommand command = new OleDbCommand();
		command.Connection = connection;
		command.CommandText = "SELECT ID, Model FROM Cars";
		command.CommandType = CommandType.Text;
		return command;
	}
	#endregion
}
#endregion

#region CustomResourceDataSource
public class CustomResourceDataSource {
	OleDbDataAdapter adapter;

	public CustomResourceDataSource() {
		this.adapter = CarsDbAdapterHelper.CreateOleDbDataAdapter();
	}

	public OleDbDataAdapter Adapter { get { return adapter; } }

	#region ObjectDataSource methods
	public DataTable SelectMethodHandler() {
		DataTable table = new DataTable();
		Adapter.Fill(table);
		return table;
	}
	#endregion
}
#endregion